home/.chezmoi.yaml.tmpl

```diff
@@ -140,6 +140,8 @@ data:
     docker:
       doRegion: nyc1
     domain: "{{ $domain }}"
+    etcd:
```

Comment:
Can we remove this? See comments below

Reply:
If we are to use the `public_services_domain` and have an `etcd` cluster available at etcd.{{ service_domain }}, I believe this can be removed. More details below.

---
The `initial_cluster_url` is the value returned by `etcdctl` on adding a new host to the cluster. The command returns 3 values, which can be set as environment variables and a new member bootstrapped. This member will join the cluster using the information in these variables.

The value of  `initial_cluster_urls` is set to the value of `ETCD_INITIAL_CLUSTER` (one of the 3 values returned).

An example is below, taken from [here](https://etcd.io/docs/v3.5/op-guide/runtime-configuration/#add-a-new-member).

```
etcdctl member add infra3 --peer-urls=http://10.0.1.13:2380
added member 9bf1b35fc7761a23 to cluster

ETCD_NAME="infra3"
ETCD_INITIAL_CLUSTER="infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380,infra2=http://10.0.1.12:2380,infra3=http://10.0.1.13:2380"
ETCD_INITIAL_CLUSTER_STATE=existing

Start the new etcd member:
export ETCD_NAME="infra3"
$ export ETCD_INITIAL_CLUSTER="infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380,infra2=http://10.0.1.12:2380,infra3=http://10.0.1.13:2380"
$ export ETCD_INITIAL_CLUSTER_STATE=existing
$ etcd --listen-client-urls http://10.0.1.13:2379 --advertise-client-urls http://10.0.1.13:2379 --listen-peer-urls http://10.0.1.13:2380 --initial-advertise-peer-urls http://10.0.1.13:2380 --data-dir %data_dir%
```

Note: The `ETCD_INITIAL_CLUSTER` value includes the _name_ and _advertised peer URLs_ of existing members and the new member. This is used by the new member to verify the configuration. 

While it is possible to use a discovery service (an internal one) to add a member (after the cluster is up and running), using [runtime configuration](https://etcd.io/docs/v3.5/op-guide/runtime-configuration) is the recommended way.

Reply:
See other comments below. We need to use the CloudFlare API to get a list of all of the subdomains for the PUBLIC_SERVICES_DOMAIN and then piece together the network like that. We can store data in TXT records, for instance and assume the master node of `etcd` for instance is available at etcd.megabyte.space.

Reply:
Hey @enggnr --- can we pull the DNS records for the PUBLIC_SERVICES_DOMAIN and return all the DNS entries that look something like `etcd.{{ .host.hostname }}.{{ .host.domain }}`? This should give all the existing nodes which we can then use to populate everything.

ETCD_NAME should just equal the `{{ .host.hostname }}`

Can you take a swing at this implementation? I think the `cfcli` should be able to provide all the functionality we need to get this working.